project4/inc/Board.hpp

```diff
@@ -0,0 +1,33 @@
+#pragma once
+#include <vector>
+
+using Fields = std::vector<std::vector<char>>;
+
+class Board
+{
+    int size_;
+    int winningNumber_;
+    Fields fields_;
+
+    bool validateCoordinates(int xCoordinate, int yCoordinate) const;
+
+    bool checkWinnerVertically(int xCoordinate,int yCoordinate, char currentPlayer) const;
+    bool checkWinnerHorizontally(int xCoordinate, int yCoordinate,char currentPlayer) const;
+    bool checkWinnerDiagonally(int xCoordinate, int yCoordinate,char currentPlayer) const;
+    bool checkWinnerAntiDiagonally(int xCoordinate, int yCoordinate,char currentPlayer) const;
+
+public:
+
+    Board() = delete;
+    Board(int size, int winningNumber);
+    Fields getFields() const;
+    void printBoard() const;
+    int getSize() const;
+    int getWinningNumber() const;
+    bool isFull() const;
+    void eraseField(int xCoordinate, int yCoordinate);
+    bool putOnBoard(int xCoordinate, int yCoordinate,char currentPlayer);
```

Comment:
', ' or ',' please apply one convention but the first one is better

Reply:
Please be awarie How the erase worka: https://en.m.wikipedia.org/wiki/Erase%E2%80%93remove_idiom. It's other C++ boiler Code solution which sadly od required